samples/template-operator/README.md

```diff
@@ -34,270 +66,511 @@ Additionally, it hides Kubernetes boilerplate code to develop fast and efficient
+  - "*"                                                                                                                                                       
+  verbs:                                                                                                                                                      
+  - "*"
+```
 
-## RBAC
-Make sure you have appropriate authorizations assigned to you controller binary, before you run in inside a cluster.
-Sample CR [controller implementation](./operator/controllers/sample_controller.go) includes rbac generation (via kubebuilder) for all resources across all API groups.
-This should certainly be adjusted according to the chart manifest resources and reconciliation types.
+_Note that this is very hard to properly protect against privilege escalation in single-cluster mode, which is one of the reasons we heavily discourage it for productive use_
 
-   ```yaml
-      // TODO: dynamically create RBACs! Remove line below.
-      //+kubebuilder:rbac:groups="*",resources="*",verbs="*"
+### Deploying a `ModuleTemplate` into the control-plane
+
+Now run the command for creating the ModuleTemplate in the Cluster.
+After this the module will be available for consumption based on the module name configured with the label `operator.kyma-project.io/module-name` on the ModuleTemplate.
+
+_WARNING: Depending on your setup against either a k3d cluster/registry, you will need to run the script in `hack/local-template.sh` before pushing the ModuleTemplate to have proper registry setup.
+(This is necessary for k3d clusters due to port-mapping issues in the cluster that the operators cannot reuse, please take a look at the [relevant issue for more details](https://github.com/kyma-project/module-manager/issues/136#issuecomment-1279542587))_
+
+```sh
+kubectl apply -f template.yaml
+```
+
+For single-cluster mode, you could use the existing Kyma Resource generated for the control-plane in `kyma.yaml` with this:
+
+```shell
+kubectl patch kyma default-kyma -n kcp-system --type='json' -p='[{"op": "add", "path": "/spec/modules", "value": [{"name": "sample" }] }]'
+```
+
+This adds your module into `.spec.modules` with a name originally based on the `"operator.kyma-project.io/module-name": "sample"` label that was generated in `template.yaml`:
+
+```yaml
+spec:
+  modules:
+  - name: sample
+```
+
+Note that of course, you can adjust the Kyma CR based on your testing scenario. For example, if you are running a Dual-Cluster Setup, you might want to enable the synchronization of the Kyma Resource into the runtime for E2E configurability.
+
+The installation should start immediately.
+
+### Debugging the Operator Ecosystem
+
+Of course, the operator ecosystem around Kyma is highly complex. So complex in fact, that it might become troublesome to debug issues in case your module is not installed.
```

Comment:
```suggestion
The operator ecosystem around Kyma is highly complex and it might become troublesome to debug issues in case your module is not installed or installed incorrectly.
```